tests/unit/einstein/test_ceph.py

```diff
@@ -0,0 +1,272 @@
+                               self.fs.rid, self.fs.pool)
+        p = subprocess.Popen(command, shell=True, stderr=subprocess.STDOUT,
+                             stdout=subprocess.PIPE)
+        output, err = p.communicate()
+        rbd_imgs = output.split()
+        self.assertListEqual(imgs, rbd_imgs)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+    def tearDown(self):
+        print('In tearDown()')
+        time.sleep(HIL_CALL_TIMEOUT)
+
+
+class TestCreateImage(unittest.TestCase):
+    """
+    Test create image
+    """
+    @trace
+    def setUp(self):
+        self.fs = Ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+    def runTest(self):
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE)
+        imgs = self.fs.list_images()
+        self.assertIn(CEPH_IMG, imgs)
+
+    def tearDown(self):
+        self.fs.remove(CEPH_IMG)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+
+class TestRemoveImage(unittest.TestCase):
+    """
+    Test remove image and remove snapshot
+    """
+    @trace
+    def setUp(self):
+        self.fs = Ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+    def runTest(self):
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE)
+        self.fs.snap_image(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertRaises(file_system_exceptions.ImageHasSnapshotException,
+                          self.fs.remove, CEPH_IMG)
+        removess = self.fs.remove_snapshot(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(removess)
+        remove = self.fs.remove(CEPH_IMG)
+        self.assertTrue(remove)
+        imgs = self.fs.list_images()
+        self.assertNotIn(CEPH_IMG, imgs)
+
+    def tearDown(self):
+        pass
+
+
+class TestClone(unittest.TestCase):
+    """
+    Clone image
+    """
+    @trace
+    def setUp(self):
+        self.fs = Ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+    def runTest(self):
+        # create image with format 2 which supports layering
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE, False, 1)
+        self.fs.snap_image(CEPH_IMG, CEPH_SNAP_IMG)
+        protectFlag = self.fs.snap_protect(CEPH_IMG, CEPH_SNAP_IMG)
+        self.assertTrue(protectFlag, "Failed to protect a snapshot!")
+        cloneFlag = self.fs.clone(CEPH_IMG, CEPH_SNAP_IMG,
+                                  CEPH_CHILD_IMG)
+        self.assertTrue(cloneFlag, "Failed to clone image")
+        children = self.fs.list_children(CEPH_IMG, CEPH_SNAP_IMG)
+        children_imgs = [child[1] for child in children if child[0] == POOL]
+        self.assertIn(CEPH_CHILD_IMG, children_imgs,
+                      "Error, cannot find child image")
+        parent_info = self.fs.get_parent_info(CEPH_CHILD_IMG)
+        # parent_info = (pool name, image name, snapshot name)
+        self.assertEqual(CEPH_IMG, parent_info[1])
+        self.assertEqual(CEPH_SNAP_IMG, parent_info[2])
```

Comment:
This assert is also not required. You have to do checks only on the clone not snap